group/views.py

```diff
@@ -68,4 +71,40 @@ def manage(request, group_id):
 def all(request):
     return render(request, "group/all.html", {
         'groups': Group.objects.all(),
-    })
\ No newline at end of file
+    })
+
+@login_required
+def join_group(request, group_id):
```

Comment:
maybe name like this ? "handle_join_leave_group" or "join_or_leave_group" (not sure lol if you can think better one ...)
since we delete too :) let's try to change something match our view action?